libdnf/base/transaction.cpp

```diff
@@ -222,6 +223,16 @@ std::string Transaction::transaction_result_to_string(const TransactionRunResult
     return {};
 }
 
+void Transaction::download() {
+    libdnf::repo::PackageDownloader downloader;
+    for (auto & tspkg : this->get_transaction_packages()) {
+        if (transaction_item_action_is_inbound(tspkg.get_action())) {
```

Comment:
We were just discussing this yesterday with @jan-kolarik . I think commandline packages should be skipped here (they should be already downloaded and inserted to commandline repo before goal resolving).
Similarly as they currently are excluded in dnf5 - https://github.com/rpm-software-management/dnf5/blob/97cdbe57c34259a0e0dc87431ac7c3e9a396aba7/dnf5/context.cpp#L281-L293

Reply:
Hey, you were faster! :smile:

Reply:
Also there are usually two usecases, one passing the `dest_dir` to the downloader, one without that using the default `cachedir`. It would be nice to unify that, maybe just add a `dest_dir` parameter to the `download` method with default empty? Just avoid using the `std::optional` on the public API :slightly_smiling_face:

Reply:
Done and done! `dest_dir` is a good call.